### PR TITLE
Fix illegal reflective access warning from protostuff

### PR DESCRIPTION
**Pull request by kaituo**
Signed-off-by: Kaituo Li kaituo@amazon.com

### Description
When using prostuff 1.7.4 under java9+ we get an Illegal reflective access warning due to protostuff 1.7.4's incompatibility with java9+.  The issue is solved in https://github.com/protostuff/protostuff/commit/b6ad96f229218a835ec3f7232fa8ee129a6d3806  and will be available in 1.8.0. Since  protostuff 1.8.0 is not released, this PR built 1.8.0 packages and made AD depend on the local packages directly. With the change, the illegal reflective access is gone.

Testing done:
1. Tested both single-stream and HCAD real-time detectors.
2. Tested backward compatibility: start a 1.2.0 cluster with protostuff 1.7.4, start detectors and wait for model checkpoints to be generated.  Reinstall AD plugin with locally packaged protostuff 1.8.0. Verified tha old model can come up and detectors can produce results.
 
### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/263
https://github.com/opensearch-project/anomaly-detection/issues/319
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -582,8 +582,10 @@ dependencies {
     compile files('lib/randomcutforest-core-2.0.1.jar')
 
     // used for serializing/deserializing rcf models.
-    compile group: 'io.protostuff', name: 'protostuff-core', version: '1.7.4'
-    compile group: 'io.protostuff', name: 'protostuff-runtime', version: '1.7.4'
+    compile files('lib/protostuff-api-1.8.0-SNAPSHOT.jar')
```
Comment (ylwu-amzn) [inline build.gradle:585]: Do you know when protostuff team plan to publish 1.8.0 to maven? We can move to public maven once it's ready.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change. Just a minor comment.

File: build.gradle (hunk 1 above)
Comment (kaituo) [reply build.gradle:585]: I don't. asked in their github repo, noboy replied.

**Review by amitgalitz — APPROVED** [review]
LGTM! Thanks for making this change

**Comment by amitgalitz** [comment]
Just realized this package was compiled under JRE 11. Meaning code can no longer be run after compiling AD with anything below (JRE 8). Is this an issue?

File: build.gradle (hunk 1 above)
Comment (dblock) [reply build.gradle:585]: I couldn't find a link for the ask, opened https://github.com/protostuff/protostuff/issues/327.

**Pull request merged**